Doc/c-api/structures.rst

```diff
@@ -62,12 +62,12 @@ the definition of all other Python objects.
    See documentation of :c:type:`PyVarObject` above.
 
 
-.. c:macro:: Py_TYPE(o)
+.. c:function:: PyTypeObject* Py_TYPE(PyObject *o)
 
-   This macro is used to access the :attr:`ob_type` member of a Python object.
-   It expands to::
+   This function is used to access the :attr:`ob_type` member of a Python object.
 
-      (((PyObject*)(o))->ob_type)
+   .. versionchanged:: 3.10
+      :c:func:`Py_TYPE()` is changed to the inline static function.
```

Comment:
You can suggest to use :c:func:`Py_SET_TYPE()` to set an object type.

You should also document the incompatible change in a https://docs.python.org/dev/whatsnew/3.9.html#id2 section: suggest to replace `Py_TYPE(obj) = new_type` with `Py_SET_TYPE(obj, new_type)`. I suggest to again move C API Changes at the bottow of the What's New in Python 3.10 document, as I did in the What's New in Python 3.9 document.

Reply:
@vstinner Got it, thanks for the kind review